public/app/features/dashboard/components/ShareModal/types.ts

```diff
@@ -1,14 +1,17 @@
 import React from 'react';
 import { PanelModel } from '@grafana/data';
-import { DashboardModel } from 'app/features/dashboard/state';
+import { DashboardModel, PanelModel as InternalPanelModel } from 'app/features/dashboard/state';
 
 export interface ShareModalTabProps {
   dashboard: DashboardModel;
   panel?: PanelModel;
   onDismiss?(): void;
 }
 
-export type ShareModalTab = React.ComponentType<ShareModalTabProps>;
+type ShareModalTabPropsWithInternalModel = ShareModalTabProps & { panel?: InternalPanelModel };
+export type ShareModalTab =
+  | React.ComponentType<ShareModalTabProps>
+  | React.ComponentType<ShareModalTabPropsWithInternalModel>;
```

Comment:
Hmm I wish we didn't need to reference the `InternalPanelModel` here, but I can see that would become hard to fix.

Reply:
It's a bit strange actually, looking at some of the other Share modal components (`ShareSnapshot`, `ShareEmbed`, and `ShareExport`) all use the internal `PanelModel` for their `panel` prop, while the `ShareModalTab` type seems to use the `@grafana/data` `PanelModel` interface. Not exactly sure why typescript didn't complain about that, but did for the component I've introduced with this PR, but I suspect it has something to do with the fact that the component I've introduced is a function component whereas the others are all class components.

Reply:
That's fine let's get this merged.